// milliseconds-parser/build/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = require("fs");
const separar = (itens, maximo) => {
    return itens.reduce((acumulador, item, indice) => {
        const grupo = Math.floor(indice / maximo);
        acumulador[grupo] = [...(acumulador[grupo] || []), item];
        return acumulador;
    }, []);
};
const SECOND = 1000;
const MINUTE = SECOND * 60;
const HOUR = MINUTE * 60;
const DAY = HOUR * 24;
const WEEK = DAY * 7;
const MONTH = WEEK * 4;
const YEAR = MONTH * 12;
class MillisecondsParser {
    constructor(language = JSON.parse(fs_1.readFileSync(__dirname + '/pt-br.json').toString())) {
        this.language = language;
        this.defaultLang = JSON.parse(fs_1.readFileSync(__dirname + '/en.json').toString());
    }
    ;
    parse(str, ignore = [" e", " and", ",", ".", "-"]) {
        var _a;
        for (const toRemove of ignore)
            str = str.replace(toRemove, "");
        const splited = separar(str.split(' '), 2);
        let total = 0;
        for (const toParse of splited) {
            let base = parseInt(toParse[0]);
            if (base <= 0)
                continue;
            const dateToParse = (_a = this.getKey(toParse[1], this.language)) !== null && _a !== void 0 ? _a : this.getKey(toParse[1], this.defaultLang);
            switch (dateToParse) {
                case "year": {
                    base *= YEAR;
                    break;
                }
                case "month": {
                    base *= MONTH;
                    break;
                }
                case "week": {
                    base *= WEEK;
                    break;
                }
                case "day": {
                    base *= DAY;
                    break;
                }
                case "hour": {
                    base *= HOUR;
                    break;
                }
                case "minute": {
                    base *= MINUTE;
                    break;
                }
                case "second": {
                    base *= SECOND;
                    break;
                }
                case "millisecond": {
                    base *= 1;
                    break;
                }
            }
            total += base;
        }
        return total;
    }
    getKey(str, language) {
        for (const key of Object.keys(language)) {
            for (const obj of language[key]) {
                if (obj.match(str))
                    return key;
            }
        }
    }
    translate(str) {
        for (const key of Object.keys(this.language)) {
            for (const obj of this.language[key]) {
                if (obj.match(str))
                    return this.language[key][this.language[key].indexOf(obj)];
            }
        }
    }
}
exports.default = MillisecondsParser;
module.exports = function (language) { return new MillisecondsParser(language); };
